packages/legend-graph/src/MetaModelUtils.ts

```diff
@@ -88,3 +89,13 @@ export const hashLambda = (
     parameters ? hashObjectWithoutSourceInformation(parameters) : '',
     body ? hashObjectWithoutSourceInformation(body) : '',
   ]);
+
+export const isCompatibleWithType = (
+  type: Type | undefined,
```

Comment:
any reason to allow the `type` for this function to be undefined? seems to specific to its usage in query builder than the actual function `name`. Returning true when its undefined seems wrong.

Reply:
I removed this, and created a query builder specific type-checking function.